lib/td/command/workflow.rb

```diff
@@ -0,0 +1,424 @@
+require 'td/helpers'
+require 'td/updater'
+require 'open3'
+require 'pathname'
+require 'time'
+require 'yaml'
+
+module TreasureData
+  module Command
+    include TreasureData::Helpers
+
+    # The workflow entrypoint command. Invokes the digdag cli, passing on any command line arguments.
+    def workflow(op, capture_output=false, check_prereqs=true)
+      if Config.apikey.nil?
+        raise ConfigError
+      end
+      check_digdag_cli unless not check_prereqs
+      cmd = [
+          java_cmd,
+          '-Dio.digdag.cli.programName=td workflow',
+          '-XX:+TieredCompilation', '-XX:TieredStopAtLevel=1', '-Xverify:none'
+      ]
+
+      FileUtils.mkdir_p digdag_tmp_dir
+      Dir.mktmpdir(nil, digdag_tmp_dir) { |wd|
+        env = {}
+        digdag_config_path = File.join(wd, 'config')
+        FileUtils.touch(digdag_config_path)
+        if Config.cl_apikey
+          # If the user passes the apikey on the command line we cannot use the digdag td.conf plugin.
+          # Instead, create a digdag configuration file with the endpoint and the specified apikey.
+          env['TD_CONFIG_PATH'] = nil
+          apikey = TreasureData::Config.apikey
+          File.write(digdag_config_path, [
+              'client.http.endpoint = https://api-workflow.treasuredata.com',
+              "client.http.headers.authorization = TD1 #{apikey}",
+              "secrets.td.apikey = #{apikey}"
+          ].join($/) + $/)
+          cmd << '-Dio.digdag.standards.td.secrets.enabled=false'
+        else
+          # Use the digdag td.conf plugin to configure wf api and apikey.
+          env['TREASURE_DATA_CONFIG_PATH'] = Config.path
+          cmd << '-Dio.digdag.standards.td.client-configurator.enabled=true'
+        end
+
+        cmd << '-jar' << digdag_cli_path
+        unless op.argv.empty?
+          cmd << '--config' << digdag_config_path
+        end
+        cmd.concat(op.argv)
+
+        unless ENV['TD_TOOLBELT_DEBUG'].nil?
+          $stderr.puts cmd.to_s
+        end
+
+        if capture_output
+          # TODO: use popen3 instead?
+          stdout_str, stderr_str, status = Open3.capture3(env, *cmd)
+          $stdout.write(stdout_str)
+          $stderr.write(stderr_str)
```

Comment:
As you say this should use popen3 because it outputs digdag's output in realtime.